pkg/ingester/lookupplan/predicate_test.go

```diff
@@ -43,14 +43,15 @@ func TestFilterCost(t *testing.T) {
 		t.Cleanup(func() { data.WriteTestCases(t, testCases) })
 	}
 
-	for _, tc := range testCases {
+	for tcIdx, tc := range testCases {
 		testName := fmt.Sprintf("%d_series_%s", tc.seriesCount, tc.matcher)
 		t.Run(testName, func(t *testing.T) {
 			pred := newPlanPredicate(ctx, tc.matcher, stats, defaultCostConfig)
 
 			cost := pred.filterCost(tc.seriesCount)
 			assert.GreaterOrEqual(t, cost, 0.0, "can't have negative costs")
 			assert.Equal(t, tc.expectedCost, cost)
+			testCases[tcIdx].expectedCost = cost
```

Comment:
Why is this necessary? E.g., it's already asserted on the previous line that they are equal?

Reply:
The assignment is for the self-updating functionality of these tests.

Reply:
Thanks - I understood from our offline chat :)